src/edgekv/ekv-service.ts

```diff
@@ -10,121 +10,336 @@ const INIT_EKV_TIMEOUT = 120000;
-  return httpEdge.getJson(`${EDGEKV_API_BASE}/networks/${network}/namespaces/${namespace}`, cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT), ekvMetrics.showNamespace).then(r => r.body).catch(err => error.handleError(err));
+  return httpEdge
+    .getJson(
+      `${EDGEKV_API_BASE}/networks/${network}/namespaces/${namespace}`,
+      cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT),
+      ekvMetrics.showNamespace
+    )
+    .then((r) => r.body)
+    .catch((err) => error.handleError(err));
 }
 
-export function updateNameSpace(network: string, namespace: string, retention, geoLocation) {
-  var body = { "namespace": namespace, "retentionInSeconds": retention, "geoLocation": geoLocation };
-  return httpEdge.putJson(`${EDGEKV_API_BASE}/networks/${network}/namespaces/${namespace}`, body, cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT), ekvMetrics.updateNamespace).then(r => r.body).catch(err => error.handleError(err));
+export function updateNameSpace(
+  network: string,
+  namespace: string,
+  retention,
+  geoLocation
+) {
+  var body = {
+    namespace: namespace,
+    retentionInSeconds: retention,
+    geoLocation: geoLocation,
+  };
+  return httpEdge
+    .putJson(
+      `${EDGEKV_API_BASE}/networks/${network}/namespaces/${namespace}`,
+      body,
+      cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT),
+      ekvMetrics.updateNamespace
+    )
+    .then((r) => r.body)
+    .catch((err) => error.handleError(err));
 }
 
 export function initializeEdgeKV() {
-  return httpEdge.putJson(`${EDGEKV_API_BASE}/initialize`, "", cliUtils.getTimeout(INIT_EKV_TIMEOUT), ekvMetrics.initialize).then(r => r.response).catch(err => error.handleError(err));
+  return httpEdge
+    .putJson(
+      `${EDGEKV_API_BASE}/initialize`,
+      '',
+      cliUtils.getTimeout(INIT_EKV_TIMEOUT),
+      ekvMetrics.initialize
+    )
+    .then((r) => r.response)
+    .catch((err) => error.handleError(err));
 }
 
 export function getInitializedEdgeKV() {
-  return httpEdge.getJson(`${EDGEKV_API_BASE}/initialize`, cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT), ekvMetrics.showInitStatus).then(r => r.response).catch(err => error.handleError(err));
+  return httpEdge
+    .getJson(
+      `${EDGEKV_API_BASE}/initialize`,
+      cliUtils.getTimeout(DEFAULT_EKV_TIMEOUT),
+      ekvMetrics.showInitStatus
+    )
+    .then((r) => r.response)
+    .catch((err) => error.handleError(err));
 }
 
-export function writeItems(network: string, namespace: string, groupId: string, itemId: string, itemList,sandboxid:string) {
+export function writeItems(
+  network: string,
+  namespace: string,
+  groupId: string,
+  itemId: string,
+  itemList,
+  sandboxid: string
+) {
   let body = itemList;
   let writeItemPath = `${EDGEKV_API_BASE}/networks/${network}/namespaces/${namespace}/groups/${groupId}/items/${itemId}`;
-  if(sandboxid){
-    writeItemPath += `?sandboxId=${sandboxid}`
+  if (sandboxid) {
+    writeItemPath += `?sandboxId=${sandboxid}`;
+  }
+
+  let request: Promise<any>;
+  if (typeof body === 'string') {
```

Comment:
Is this the section that fixes the bug in EW-12480?

It's good to see some clean up done as well but I just want to confirm.

Reply:
Yes, this is the fix.